addons/mail/static/src/discuss/core/common/composer_patch.js

```diff
@@ -3,10 +3,10 @@ import { patch } from "@web/core/utils/patch";
 
 patch(Composer.prototype, {
     get allowUpload() {
-        const thread = this.thread ?? this.message.thread;
+        const thread = this.thread ?? this.message?.thread;
         return (
             super.allowUpload &&
-            (thread.model !== "discuss.channel" ||
+            (thread?.model !== "discuss.channel" ||
```

Comment:
`message` may not be available due to the addition of `return await` to the `edit method` patch in `discuss/core/common` depending on async timing.